suitcase/tiff_series/__init__.py

```diff
@@ -0,0 +1,228 @@
+
+    >>> export(gen, '', '{start[plan_name]}-{start[motors]}-')
+
+    Include the experiment's start time formatted as YY-MM-DD_HH-MM.
+
+    >>> export(gen, '', '{start[time]:%Y-%m-%d_%H:%M}-')
+
+    Place the files in a different directory, such as on a mounted USB stick.
+
+    >>> export(gen, '/path/to/my_usb_stick')
+    """
+    with Serializer(directory, file_prefix,
+                    bigtiff=bigtiff,
+                    byteorder=byteorder,
+                    imagej=imagej,
+                    **kwargs) as serializer:
+        for item in gen:
+            serializer(*item)
+
+    return serializer.artifacts
+
+
+# Note that this is a subclass of tiff_stack.Serializer to reduce code
+# duplication.
+
+
+class Serializer(tiff_stack.Serializer):
+    """
+    Serialize a stream of documents to a series of TIFF files.
+
+    This creates a file named:
+    ``<directory>/<file_prefix>{stream_name}-{field}-{image_number}.tiff``
+    for every Event stream and field that contains 2D 'image like' data.
+
+    .. warning::
+
+        This process explicitly ignores all data that is not 2D and does not
+        include any metadata in the output file.
+
+
+    .. note::
+
+        This can alternatively be used to write data to generic buffers rather
+        than creating files on disk. See the documentation for the
+        ``directory`` parameter below.
+
+    Parameters
+    ----------
+    directory : string, Path or Manager.
+        For basic uses, this should be the path to the output directory given
+        as a string or Path object. Use an empty string ``''`` to place files
+        in the current working directory.
+
+        In advanced applications, this may direct the serialized output to a
+        memory buffer, network socket, or other writable buffer. It should be
+        an instance of ``suitcase.utils.MemoryBufferManager`` and
+        ``suitcase.utils.MultiFileManager`` or any object implementing that
+        interface. See the suitcase documentation
+        (http://nsls-ii.github.io/suitcase) for details.
+
+    file_prefix : str, optional
+        The first part of the filename of the generated output files. This
+        string may include templates as in
+        ``{start[proposal_id]}-{start[sample_name]}-``, which are populated
+        from the RunStart (start), EventDescriptor (descriptor) or Event
+        (event) documents. The default value is ``{start[uid]}-`` which is
+        guaranteed to be present and unique. A more descriptive value depends
+        on the application and is therefore left to the user.
+
+    bigtiff : boolean, optional
+        Passed into ``tifffile.TiffWriter``. Default False.
+
+    byteorder : string or None, optional
+        Passed into ``tifffile.TiffWriter``. Default None.
+
+    imagej: boolean, optional
+        Passed into ``tifffile.TiffWriter``. Default False.
+
+    **kwargs : kwargs
+        kwargs to be passed to ``tifffile.TiffWriter.save``.
+    """
```

Comment:
This docstring is an identical copy of the one from the `export` function above. Can it be written in one place and then reused? Will `Serializer.__doc__ = export.__doc__` work in this case?

Reply:
Unfortunately no, because of the `gen` parameter that `export` requires.

Reply:
resolving as @danielballan answered the question